src/test/java/io/jenkins/plugins/analysis/warnings/FlexiblePublishITest.java

```diff
@@ -0,0 +1,159 @@
+package io.jenkins.plugins.analysis.warnings;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.jenkins_ci.plugins.flexible_publish.ConditionalPublisher;
+import org.jenkins_ci.plugins.flexible_publish.FlexiblePublisher;
+import org.jenkins_ci.plugins.run_condition.BuildStepRunner.Run;
+import org.jenkins_ci.plugins.run_condition.core.AlwaysRun;
+import org.junit.Test;
+
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+
+import edu.hm.hafner.analysis.Report.IssueFilterBuilder;
+
+import hudson.model.FreeStyleBuild;
+import hudson.model.FreeStyleProject;
+import hudson.model.HealthReport;
+import hudson.model.Item;
+import hudson.tasks.BuildStep;
+
+import io.jenkins.plugins.analysis.core.filter.ExcludeFile;
+import io.jenkins.plugins.analysis.core.filter.RegexpFilter;
+import io.jenkins.plugins.analysis.core.model.ResultAction;
+import io.jenkins.plugins.analysis.core.model.AnalysisResult;
+import io.jenkins.plugins.analysis.core.steps.IssuesRecorder;
+import io.jenkins.plugins.analysis.core.testutil.IntegrationTestWithJenkinsPerSuite;
+import io.jenkins.plugins.analysis.warnings.recorder.pageobj.DetailsTab;
+import io.jenkins.plugins.analysis.warnings.recorder.pageobj.DetailsTab.TabType;
+import io.jenkins.plugins.analysis.warnings.recorder.pageobj.IssueRow;
+import io.jenkins.plugins.analysis.warnings.recorder.pageobj.IssuesTable;
+
+import static org.assertj.core.api.Assertions.*;
+
+/**
+ * Test the flexible publish plugin in combination with the warnings-ng-plugin.
+ *
+ * @author Tobias Redl
+ * @author Andreas Neumeier
+ */
+public class FlexiblePublishITest extends IntegrationTestWithJenkinsPerSuite {
+    private static final String JAVA_FILE = "java2Warnings.txt";
+    private static final String JAVA_FILE2 = "java-start.txt";
+    private static final String JAVA_PATTERN = "**/*.txt";
+    private static final String TOOL_ID = "sampleId1";
+    private static final String TOOL_ID2 = "sampleId2";
+
+    /**
+     * Test that the same tool can be used twice with different configuration.
+     */
+    @Test
+    public void shouldAnalyseJavaTwiceWithHealthReport() {
+        FreeStyleProject project = createFreeStyleProject();
+        copySingleFileToWorkspace(project, JAVA_FILE, JAVA_FILE);
+        copySingleFileToWorkspace(project, JAVA_FILE2, JAVA_FILE2);
+
+        IssuesRecorder publisher = constructJavaIssuesRecorder(JAVA_FILE, TOOL_ID, 1, 9);
+        IssuesRecorder publisher2 = constructJavaIssuesRecorder(JAVA_PATTERN, TOOL_ID2, 1, 3);
+
+        project.getPublishersList().add(new FlexiblePublisher(Arrays.asList(
+                constructConditionalPublisher(publisher),
+                constructConditionalPublisher(publisher2)
+        )));
+
+        hudson.model.Run<?, ?> run = buildSuccessfully(project);
```

Comment:
use imports

Reply:
@uhafner we can not use and import for `hudson.model.Run` because there is another one with the same name: `import org.jenkins_ci.plugins.run_condition.BuildStepRunner.Run;`